packages/netlify-cms-core/src/backend.ts

```diff
@@ -349,7 +275,7 @@ export class Backend {
   async entryExist(collection: Collection, path: string, slug: string) {
     const unpublishedEntry =
       this.implementation.unpublishedEntry &&
-      (await this.implementation.unpublishedEntry(collection, slug).catch(error => {
+      (await this.implementation.unpublishedEntry(collection.get('name'), slug).catch(error => {
```

Comment:
The backends only need the collection name. I've aligned all the backends to just receive the relevant arguments instead of the collection immutable object (we have some places we were passing a collection object and other places we were passing a collection string which was confusing).